### PR TITLE
AIEV2 driver support

### PR DESCRIPTION
**Pull request by larry9523**
This PR is to switch XRT Graph API to use AIE V2 driver. Major changes include
1) When loading xclbin, zocl will request an AIE partition dev
2) Provide an IOCTL to require AIE partition fd. 
    Any user process who wants to initialize AIE instance and call AIE driver user space API need to talk to zocl driver to require an fd
3) In libxrt_core, xrt will call AIE V2 driver API instead of calling AIE V1 API.

@pgschuey I temporarily comment out a few code in aie_plugin.cpp to get compiling through. Please make changes accordingly to use v2 API.
@mamin506 Please review. Especially the aie_pdi_only part. Hopefully, I won't break anything.
@stsoe Please review the code. This PR can not be merged for now. But once petalinux yocto ai-engien-driver recipe is moved to point to aiev2 repository, this PR needs to be merged ASAP to keep petalinux going. The pipeline will be failing until petalinux move to aiev2 recipe
@chvamshi-xilinx Once it merged, please help to submit the commit id to petalinux yocto xrt recipe ASAP.

Work to be done
xrtResetAIEArray will not work after moving to V2. This is will be my next PR
If do a rmmod zocl and insmod zocl, talking to AIE tile will cause SError. This will be my next PR.

### CODE REVIEW

File: src/runtime_src/core/edge/drm/zocl/zocl_aie.c [hunk 1]
```diff
@@ -0,0 +1,132 @@
+/* SPDX-License-Identifier: GPL-2.0 OR Apache-2.0 */
+/*
+ * A GEM style (optionally CMA backed) device manager for ZynQ based
+ * OpenCL accelerators.
+ *
+ * Copyright (C) 2016-2020 Xilinx, Inc. All rights reserved.
```
Comment (mamin506) [inline src/runtime_src/core/edge/drm/zocl/zocl_aie.c:6]: 2020 Xilinx?

Comment (larry9523) [reply src/runtime_src/core/edge/drm/zocl/zocl_aie.c]: Fixed

**Comment by larry9523** [comment]
> This PR cannot be merged until the AIE profiling plugin (aie_plugin.cpp) is moved to the V2 API. As is currently implemented in this PR, this will break the profiling feature. The proper APIs to access the counters must be in place in V2, and the gathering of the proper information to call tose APIs need to be added as well.

Yes, it will break the profiling feature. But the AIE driver is migrating to V2. If we don't merge, 1) petalinux build will be broken, 2) XRT edge is broken. That is why I just commented out the code calling into AIE V1 driver in aie_plugin.cpp.
@pgschuey if you have a code ready for V2, you can either send to me and I can merge it to this PR, or you can submit a PR now or as soon as this PR is merged. So that we can minimize the broken time of profiling feature.

**Comment by larry9523** [comment]
> > This PR cannot be merged until the AIE profiling plugin (aie_plugin.cpp) is moved to the V2 API. As is currently implemented in this PR, this will break the profiling feature. The proper APIs to access the counters must be in place in V2, and the gathering of the proper information to call tose APIs need to be added as well.
> 
> Yes, it will break the profiling feature. But the AIE driver is migrating to V2. If we don't merge, 1) petalinux build will be broken, 2) XRT edge is broken, and profiling feature is broken anyway. That is why I just commented out the code calling into AIE V1 driver in aie_plugin.cpp.
> @pgschuey if you have a code ready for V2, you can either send to me and I can merge it to this PR, or you can submit a PR now or as soon as this PR is merged. So that we can minimize the broken time of profiling feature.

Merged the new profiling code to use v2 driver. @pgschuey

**Comment by dayeh-xilinx** [comment]
test this please

**Comment by abalasa** [comment]
retest this please

**Comment by larry9523** [comment]
@chvamshi-xilinx should we merge this now?

**Pull request merged**